src/core/IPosisProcess.d.ts

```diff
@@ -3,6 +3,8 @@ interface IPosisProcess {
     imageName: string; // image name (maps to constructor)
     id: PosisPID; // ID
     parentId: PosisPID; // Parent ID
-    log: IPosisLogger; // Logger 
+    log: IPosisLogger; // Logger
+    // For querying extension interfaces (instead of tying ourselves to "levels")
+    queryPosisInterface<T extends keyof PosisInterfaces>(interfaceId: T): PosisInterfaces[T] | undefined;
```

Comment:
Not a fan of this, since we cannot use a base process for processes, injecting a function while possible seems like the wrong way to do it.